crates/algod_client/src/models/signed_txn_in_block.rs

```diff
@@ -0,0 +1,96 @@
+/*
+ * Algod REST API.
+ *
+ * API endpoint for algod operations.
+ *
+ * The version of the OpenAPI document: 0.0.1
+ * Contact: contact@algorand.com
+ * Generated by: Rust OpenAPI Generator
+ */
+
+use crate::models;
+#[cfg(not(feature = "ffi_uniffi"))]
+use algokit_transact::SignedTransaction as AlgokitSignedTransaction;
+use serde::{Deserialize, Serialize};
+use serde_with::{Bytes, serde_as};
+
+#[cfg(feature = "ffi_uniffi")]
+use algokit_transact_ffi::SignedTransaction as AlgokitSignedTransaction;
+
+use algokit_transact::AlgorandMsgpack;
+
+use crate::models::UnknownJsonValue;
+
+/// SignedTxnInBlock is a SignedTransaction with additional ApplyData and block-specific metadata.
+#[serde_as]
+#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(feature = "ffi_uniffi", derive(uniffi::Record))]
+pub struct SignedTxnInBlock {
+    /// [sig] Standard Ed25519 signature.
+    #[serde_as(as = "Option<Bytes>")]
+    #[serde(rename = "sig", skip_serializing_if = "Option::is_none")]
+    pub signature: Option<Vec<u8>>,
+    /// [msig] Multisignature signature.
+    #[serde(rename = "msig", skip_serializing_if = "Option::is_none")]
+    pub multisignature: Option<UnknownJsonValue>,
+    /// [lsig] Logic signature (program signature).
+    #[serde(rename = "lsig", skip_serializing_if = "Option::is_none")]
+    pub logic_signature: Option<UnknownJsonValue>,
+    /// [txn] The transaction that was signed.
+    #[serde(rename = "txn", skip_serializing_if = "Option::is_none")]
+    pub transaction: Option<UnknownJsonValue>,
+    /// [sgnr] The signer, if signing delegated to different key.
+    #[serde(rename = "sgnr", skip_serializing_if = "Option::is_none")]
+    pub auth_address: Option<String>,
+    /// [ca] Rewards applied to close-remainder-to account.
+    #[serde(rename = "ca", skip_serializing_if = "Option::is_none")]
+    pub closing_amount: Option<u64>,
+    /// [aca] Asset closing amount.
+    #[serde(rename = "aca", skip_serializing_if = "Option::is_none")]
+    pub asset_closing_amount: Option<u64>,
+    /// [rs] Sender rewards.
+    #[serde(rename = "rs", skip_serializing_if = "Option::is_none")]
+    pub sender_rewards: Option<u64>,
+    /// [rr] Receiver rewards.
+    #[serde(rename = "rr", skip_serializing_if = "Option::is_none")]
+    pub receiver_rewards: Option<u64>,
+    /// [rc] Close rewards.
+    #[serde(rename = "rc", skip_serializing_if = "Option::is_none")]
+    pub close_rewards: Option<u64>,
+    /// [dt] State changes from app execution.
+    #[serde(rename = "dt", skip_serializing_if = "Option::is_none")]
+    pub eval_delta: Option<UnknownJsonValue>,
```

Comment:
Don't we already have EvalDelta model? Or is the model different on this endpoint?

Reply:
Yes, we already have the TEAL value-level EvalDelta model. But the block endpoint’s dt is transactions.ApplyData.EvalDelta, a different, larger struct that includes GlobalDelta/LocalDeltas of StateDelta (which use the TEAL value EvalDelta entries), plus logs and inner transactions. I think its better defer to a raw msgpack bytes for now and add an injected typed model later if we need it.